src/app/services/actions/query-action-creator-util.ts

```diff
@@ -1,158 +1,200 @@
+  };
+
+  const options: IRequestOptions = { method: query.selectedVerb, headers };
+  return { graphUrl, options };
+}
+
+export function authenticatedRequest(
+  dispatch: Function,
+  query: IQuery,
+  devxApiUrl?: string,
+  permissionModeType?: PERMISSION_MODE_TYPE,
+  scopes: string[] = DEFAULT_USER_SCOPES.split(' ')
+) {
+  if (permissionModeType === PERMISSION_MODE_TYPE.TeamsApp) {
+    return makeAppRequest(query.selectedVerb, devxApiUrl as string)(dispatch, query);
+  } else {
+    return makeDelegatedRequest(query.selectedVerb, scopes)(dispatch, query);
+  }
+}
+
+export function isImageResponse(contentType: string | undefined) {
+  if (!contentType) {
+    return false;
+  }
+  return (
+    contentType === 'application/octet-stream' || contentType.includes('image/')
+  );
+}
+
+export function isBetaURLResponse(json: any) {
+  return !!json?.account?.[0]?.source?.type?.[0];
+}
+
+export function getContentType(headers: Headers) {
+  const contentType = headers.get('content-type');
+  if (contentType) {
+    const delimiterPos = contentType.indexOf(';');
+    if (delimiterPos !== -1) {
+      return contentType.substr(0, delimiterPos);
+    } else {
+      return contentType;
+    }
+  }
+}
+
+export function parseResponse(response: any, respHeaders: any): Promise<any> {
+  if (response && response.headers) {
+    response.headers.forEach((val: any, key: any) => {
+      respHeaders[key] = val;
+    });
+
+    const contentType = getContentType(response.headers);
+    switch (contentType) {
+      case ContentType.Json:
+        return response.json();
+
+      case ContentType.XML:
+      case ContentType.HTML:
+      case ContentType.TextPlain:
+        return response.text();
+
+      default:
+        return response;
+    }
+  }
+  return response;
+}
+
+function getSampleHeaders(query: IQuery): any {
+  const sampleHeaders: any = {};
+  sampleHeaders.SdkVersion = 'GraphExplorer/4.0';
+
+  if (query.sampleHeaders && query.sampleHeaders.length > 0) {
+    query.sampleHeaders.forEach((header) => {
+      sampleHeaders[header.name] = header.value;
+    });
+  }
+
+  return sampleHeaders;
+}
+
+const makeAppRequest = (httpVerb: string, devxapiUrl: string): Function => {
+  return async (dispatch: Function, query: IQuery) => {
+    const headers: any = getSampleHeaders(query);
+
+    // Adding user delegated token to request header
+    const token = await authProvider.getAccessToken();
+    headers.Authorization = `Bearer ${token}`;
+
+    const options: IRequestOptions = {
+      method: httpVerb,
+      headers
+    }
+
+    if (httpVerb !== 'GET' && !!query.sampleBody) {
+      options.body = JSON.stringify(query.sampleBody);
+    }
+
+    dispatch(queryRunningStatus(true));
+
+    const graphEndpoint = query.sampleUrl.replace(GRAPH_URL, '');
+    const requestUrl = `${devxapiUrl}/graphproxy${graphEndpoint}`;
+
+    const response = await fetch(requestUrl, options);
+    if (!response.ok) {
+      throw response;
+    }

```

Comment:
Why are we throwing the response?

Reply:
Oops, yes we dont need to throw the error here. Thank you